knowledge_base/Environment-Variables.md

```diff
@@ -0,0 +1,401 @@
+
+API key for Ethereum data.
+
+## FLAG_COMMUNITY_HOMEPAGE
+
+Boolean toggle to display side-wide homepage feature for communities. Temporary flag for 2.0 work.
+
+## FLAG_PROPOSAL_TEMPLATES
+
+Boolean toggle to display side-wide sidebar proposal templates. Temporary flag for 2.0 work.
+
+## FLAG_SIDEBAR_TOGGLE
+
+Boolean toggle to display site-wide discussions homepage sidebar. Temporary flag for 2.0 work.
+
+## HEROKU_APP_NAME
+
+Set in Heroku; the respective app name. Required when using the DataDog PSQL integration.
+
+## IS_CI
+
+Used and defined on-the-fly in e2e scripts.
+
+Owner: Kurtis Assad.
+
+## JWT_SECRET
+
+Required in production. The JWT seed secret that is used to generate all user JWTs.
+
+## MAGIC_API_KEY
+
+Secret API key for Magic login. Contact Jake Naviasky or Graham Johnson for access.
+
+## MAGIC_DEFAULT_CHAIN
+
+Default chain for Magic login; as of 231212, we use `ethereum`.
+
+## MAGIC_PUBLISHABLE_KEY
+
+Publishable API key for Magic login; as of 231212, development uses `pk_live_EF89AABAFB87D6F4`.
+
+## MAGIC_SUPPORTED_BASES
+
+Chain bases supported for Magic login; as of 231212, we use  `substrate,ethereum`.
+
+## MIXPANEL_DEV_TOKEN
+
+Mixpanel analytics tracking token for development work. Reach out to a lead to request a token.
+
+## MIXPANEL_PROD_TOKEN
+
+Mixpanel analytics tracking token for our live production site.
+
+## NEXT_PUBLIC_RSA_PRIVATE_KEY
+
+<!-- Introduced in #2096 but never documented. If you have information about this token, please contribute. Otherwise, this entry is flagged for removal, and the referrent code for refactoring. -->
+
+## NEXT_PUBLIC_RSA_PUBLIC_KEY
+
+<!-- Introduced in #2096 but never documented. If you have information about this token, please contribute. Otherwise, this entry is flagged for removal, and the referrent code for refactoring. -->
+
+## NO_CLIENT
+
+If `true`, disables the front-end build.
+
+## NO_GLOBAL_ACTIVITY_CACHE
+
+If `true`, disables the initialization of `globalActivityCache.ts` from server.
+
+## NO_PRERENDER
+
+In a production environment, prerender is only run from `commonwealth/server.ts` if this flag is false or blank.
+
+## NO_SSL
+
+Used and defined on-the-fly for the `start-external-webpack` package.json script.
+
+## NO_TOKEN_BALANCE_CACHE
+
+If `true`, the token balance cache is not initialized.
+
+## NODE_ENV
+
+The current environment; set to `production` or `development`. Upstream of how we handle several other env vars.
+
+## PGPASSWORD
+
+Postgres DB password (by default, `edgeware`). Bypasses usual password prompt for local DB commands.
+
+## PORT
+
+Localhost port location, default value `8080`.
+
+## PROCFILE
+
+Set in all of our Heroku apps that use a separate procfile. Specifies the path from the root of the repo to the Procfile to use for the deployed app. It is never needed locally but it must be set manually on Heroku and is required in production. Presently set to `packages/chain-events/Procfile`
+
+## RABBITMQ_API_URI
```

Comment:
This variable will default to the URI of a local RabbitMQ API server. This allows the use of the message publishing script to publish a message to a local RabbitMQ instance.